rust/lance/src/catalog/catalog_trait.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright The Lance Authors
+
+use crate::catalog::dataset_identifier::DatasetIdentifier;
+use crate::catalog::namespace::Namespace;
+use crate::dataset::Dataset;
+use std::collections::{HashMap, HashSet};
+
+pub trait Catalog {
+    /// Initialize the catalog.
+    fn initialize(&self, name: &str, properties: &HashMap<&str, &str>) -> Result<(), String>;
+
+    /// List all datasets under a specified namespace.
+    fn list_datasets(&self, namespace: &Namespace) -> Vec<DatasetIdentifier>;
+
+    /// Create a new dataset in the catalog.
+    fn create_dataset(
```

Comment:
I think using `register_dataset` interface to register an existed lance dataset is also needed.